plugin/contrib/mesos/pkg/executor/executor_test.go

```diff
@@ -195,3 +207,163 @@ func TestSuicide_WithTasks(t *testing.T) {
 		glog.Infoln("jumps verified") // glog so we get a timestamp
 	}
 }
+
+func TestExecutorRegister(t *testing.T) {
+	flag.Lookup("v").Value.Set(fmt.Sprint(*test_v))
+
+	mockDriver := MockExecutorDriver{}
+	executor := NewTestKubernetesExecutor()
+
+	executor.Init(mockDriver)
+	executor.Registered(mockDriver, nil, nil, nil)
+
+	assert.Equal(t, executor.isConnected(), true, "executor should be connected")
+	mockDriver.AssertExpectations(t)
+}
+
+func TestExecutorReregister(t *testing.T) {
+	flag.Lookup("v").Value.Set(fmt.Sprint(*test_v))
+
+	mockDriver := MockExecutorDriver{}
+	executor := NewTestKubernetesExecutor()
+
+	executor.Init(mockDriver)
+	executor.Registered(mockDriver, nil, nil, nil)
+
+	assert.Equal(t, executor.isConnected(), true, "executor should be connected")
+	mockDriver.AssertExpectations(t)
+}
+
+func TestExecutorDisconnect(t *testing.T) {
+	flag.Lookup("v").Value.Set(fmt.Sprint(*test_v))
+
+	mockDriver := MockExecutorDriver{}
+	executor := NewTestKubernetesExecutor()
+
+	executor.Init(mockDriver)
+	executor.Registered(mockDriver, nil, nil, nil)
+	executor.Disconnected(mockDriver)
+
+	assert.Equal(t, executor.isConnected(), false, "executor should not be connected after Disconnected")
+	mockDriver.AssertExpectations(t)
+}
+
+func TestExecutorLaunchAndKillTask(t *testing.T) {
+	flag.Lookup("v").Value.Set(fmt.Sprint(*test_v))
+
+	mockDriver := MockExecutorDriver{}
+	config := Config{
+		Docker:    dockertools.ConnectToDockerOrDie("fake://"),
+		Updates:   make(chan interface{}, 1024),
+		APIClient: client.NewOrDie(&client.Config{Host: "fakehost", Version: testapi.Version()}),
+	}
+	executor := New(config)
+
+	executor.Init(mockDriver)
+	executor.Registered(mockDriver, nil, nil, nil)
+
+	pod := newTestPod(1)
+	podTask, err := podtask.New(api.NewDefaultContext(), "",
+		*pod, &mesosproto.ExecutorInfo{})
+	assert.Equal(t, err, nil, "must be able to create a task from a pod")
+
+	taskInfo := podTask.BuildTaskInfo()
```

Comment:
I would put a comment line above each block, describing the "story" of the test.